controllers/numaresourcesscheduler_controller_test.go

```diff
@@ -417,7 +410,55 @@ var _ = ginkgo.Describe("Test NUMAResourcesScheduler Reconcile", func() {
 			_, err := reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: key})
 			gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
-			expectCacheParams(reconciler.Client, depmanifests.CacheResyncAutodetect, depmanifests.ForeignPodsDetectAll)
+			expectCacheParams(reconciler.Client, depmanifests.CacheResyncAutodetect, depmanifests.ForeignPodsDetectAll, depmanifests.CacheInformerDedicated)
+		})
+
+		ginkgo.It("should configure by default the informerMode to be Dedicated", func() {
+			key := client.ObjectKeyFromObject(nrs)
+			_, err := reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: key})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			expectCacheParams(reconciler.Client, depmanifests.CacheResyncAutodetect, depmanifests.CacheResyncOnlyExclusiveResources, depmanifests.CacheInformerDedicated)
+
+		})
+
+		ginkgo.It("should allow to change the informerMode to Shared", func() {
+			nrs := nrs.DeepCopy()
+			informerMode := nropv1.SchedulerInformerShared
+			nrs.Spec.SchedulerInformer = &informerMode
+			gomega.Eventually(func() bool {
+				if err := reconciler.Client.Update(context.TODO(), nrs); err != nil {
+					klog.Warningf("failed to update the scheduler object; err: %v", err)
+					return false
+				}
+				return true
+			}, 30*time.Second, 5*time.Second).Should(gomega.BeTrue())
+
+			key := client.ObjectKeyFromObject(nrs)
+			_, err := reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: key})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			expectCacheParams(reconciler.Client, depmanifests.CacheResyncAutodetect, depmanifests.CacheResyncOnlyExclusiveResources, depmanifests.CacheInformerShared)
+
+		})
+
+		ginkgo.It("should allow to change the informerMode to Dedicated", func() {
```

Comment:
this test makes sense only if we
1. start with informerMode dedicated
2. we can correctly change to shared
3. we can then correctly change it back again to dedicated

if you followed another test which doesn't do that, the other test is wrong and needs to be fixed in another PR